robottelo/utils/virtwho.py

```diff
@@ -87,31 +88,52 @@ def runcmd(cmd, system=None, timeout=600000, output_format='base'):
     return result.status, result.stdout.strip()
 
 
-def register_system(system, activation_key=None, org='Default_Organization', env='Library'):
+def register_system(
+    system, activation_key=None, org='Default_Organization', env='Library', target=None
+):
     """Return True if the system is registered to satellite successfully.
 
     :param dict system: system account used by ssh to connect and register.
     :param str activation_key: the activation key will be used to register.
-    :param str org: Which organization will be used to register.
+    :param str org: Which organization will be used to register (org label).
     :param str env: Which environment will be used to register.
+    :param target: Satellite object for registration.
     :raises: VirtWhoError: If failed to register the system.
     """
-    runcmd('subscription-manager unregister', system)
-    runcmd('subscription-manager clean', system)
-    runcmd('rpm -qa | grep katello-ca-consumer | xargs rpm -e |sort', system)
-    runcmd(
-        f'rpm -ihv http://{settings.server.hostname}/pub/katello-ca-consumer-latest.noarch.rpm',
-        system,
+    # Get the guest hostname to check if it's already registered
+    guest_name, _ = runcmd('hostname', system=system)
+
+    if existing_hosts := Host.list({'search': guest_name}):
+        for host in existing_hosts:
+            Host.delete({'id': host['id']})
+
+    # Create ContentHost object from system dict
+    contenthost = ContentHost(
+        hostname=system['hostname'], auth=(system['username'], system['password'])
     )
-    cmd = f'subscription-manager register --org={org} --environment={env} --force '
-    if activation_key is not None:
-        cmd += f'--activationkey={activation_key}'
-    else:
-        cmd += f'--username={settings.server.admin_username} --password={settings.server.admin_password}'
-    ret, stdout = runcmd(cmd, system)
-    if ret == 0 or "system has been registered" in stdout:
-        return True
-    raise VirtWhoError(f'Failed to register system: {system}')
+
+    # Use the ContentHost's register() method with global registration
+    try:
+        # If org is a string (label), look up the organization object
+        org_obj = None
+        if isinstance(org, str):
+            org_obj = entities.Organization().search(query={'search': f'label={org}'})[0]
+        else:
+            org_obj = org
+
+        result = contenthost.register(
+            org=org_obj,  # org object
+            loc=None,  # location object or None
+            activation_keys=activation_key,  # activation key name (required for global registration)
+            target=target,  # target satellite object
+            force=True,
+        )
+
+        if result.status == 0:
+            return True
+        raise VirtWhoError(f'Failed to register system: {system}\n {result}')
```

Comment:
This would return specific context related to error only.
```suggestion
        raise VirtWhoError(f'Failed to register system: {system}\n {result.error}')
```